layouts/fastly/helix.vcl

```diff
@@ -1213,10 +1213,17 @@ sub hlx_type_content {
 sub hlx_type_preflight {
   set req.http.X-Trace = req.http.X-Trace + "; hlx_type_preflight";
 
+  call hlx_owner;
+  call hlx_repo;
+  call hlx_ref;
+
   # get it from OpenWhisk (for now, we will support other backends later)
   set req.backend = F_AdobeRuntime;
 
-  set req.http.X-Backend-URL = {"const:preflight"};
+  set req.http.X-Backend-URL = {"const:preflight"}
+      + "?owner=" + req.http.X-Owner
+      + "&repo=" + req.http.X-Repo
+      + "&ref=" + req.http.X-Ref;
```

Comment:
Don't you think it would be enough to have this in the headers? That way the cache efficiency stays high, as you'd only `Vary` on the headers you actually need?

Reply:
but it's different for each repository picked repository. if we add owner/repo/ref as headers, we also need to vary on all of them..

I also thought of using the x-helix-forwarded-url header, and compute the coords; but this would then be less robust.

Reply:
The advantage of using headers and vary is that you get exactly the granularity you need, `x-helix-forwarded-url` would be too fine grained.